cylc/flow/data_store_mgr.py

```diff
@@ -299,6 +306,7 @@ class DataStoreMgr:
         'min_point',
         'parents',
         'pool_points',
+        'publish_deltas',
```

Comment:
I should have picked this up previously, but we probably don't need `__slots__` in this class.  As I understand it, it's only a significant advantage (memory footprint) for classes that instantiate a large number of instances, such as task proxy and cycling-related objects in cylc-flow.

Reply:
Official docs on explain it as such, anyway:
https://docs.python.org/3/glossary.html#term-slots

> best reserved for rare cases where there are large numbers of instances in a memory-critical application.

Reply:
Protobuf messages are class instances, does this count?

Reply:
It might help if the Protobuf message class itself had slots, but this is the `DataStoreMgr` class, which only gets instantiated once, right?

Reply:
i.e. wasted/spare memory in a small attributes dict doesn't matter much for a single instance, but it might for a very large number of instances.

Reply:
Roger that, will remove slots (although it was helpful for the memory profiler to iterate through)

Reply:
`__slots__` are unnecessary for classes like this that don't get initiated often, however, they are also harmless.

Reply:
Yeah no biggie, but still, I don't think we should use __slots__ unnecessarily. There is at least a small maintenance cost, and it makes the class slightly harder to use (in experimental mode, at least).